### PR TITLE
chore: enhance create superuser script

### PR DESCRIPTION
**Pull request by reidliu41**
# [AAP-XXXXX] Brief Title Describing the Change

## Description

- add a `poetry` command check
- add the `usage` for the invalid flag, even invalid flag should show the usage, make a better `UX`.

Closes #XXX (if applicable)
Jira: [AAP-XXXXX](https://jira.aap.com/browse/AAP-XXXXX) (if applicable)

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change
- [ ] CI change

## Self-Review Checklist

- [ ] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment
- [ ] I have run the linters and test suite locally
- [ ] I have tested the changes on integration environment

### Steps to Test

1.
2.
3.

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX | <https://github.com/example/repo/pull/XXX>

### CODE REVIEW

File: scripts/create_superuser.sh [hunk 1]
```diff
@@ -26,6 +26,11 @@ usage() {
 }
 
 create_user() {
+    if ! command -v poetry &>/dev/null; then
+        log-warn "poetry is not installed, please install poetry(version >= 1.4.0) first."
```
Comment (Alex-Izquierdo) [inline scripts/create_superuser.sh:30]: We upgrade to poetry 2.x, probably not an issue for `run` sub command but I would require it anyway for safeness and consistency. 
```suggestion
        log-warn "poetry is not installed, please install poetry(version >= 2.0 ) first."
```

Comment (Alex-Izquierdo) [reply scripts/create_superuser.sh]: Thanks for attending my suggestions. And thanks for your contribution.
